theano/gof/cmodule.py

```diff
@@ -1969,6 +1974,14 @@ def compile_str(module_name, src_code, location=None,
         else:
             cmd.extend(preargs)
         cmd.extend('-I%s' % idir for idir in include_dirs)
+        if hide_symbols and sys.platform != 'win32':
+            # This has been available since gcc 4.0 so we suppose it
+            # is always available. We pass it here since it
+            # significantly reduces the size of the symbol table for
+            # the objects we want to share. This in turns leads to
+            # improved loading times on most platforms (win32 is
+            # different, as usual).
+            cmd.append('-fvisibility=hidden')
```

Comment:
About windows, did you test it? If not, can you do it? Otherwise, I'll see
if we can do it in the lab.

On Wed, Apr 15, 2015 at 2:21 PM, abergeron <notifications@github.com> wrote:

> In theano/gof/cmodule.py
> <https://github.com/Theano/Theano/pull/2765#discussion_r28447443>:
>
> > @@ -1969,6 +1974,14 @@ def compile_str(module_name, src_code, location=None,
> >          else:
> >              cmd.extend(preargs)
> >          cmd.extend('-I%s' % idir for idir in include_dirs)
> > +        if hide_symbols and sys.platform != 'win32':
> > +            # This has been available since gcc 4.0 so we suppose it
> > +            # is always available. We pass it here since it
> > +            # significantly reduces the size of the symbol table for
> > +            # the objects we want to share. This in turns leads to
> > +            # improved loading times on most platforms (win32 is
> > +            # different, as usual).
> > +            cmd.append('-fvisibility=hidden')
>
> Because otherwise the hide_symbols parameter would have to be on
> compile_args and I think it fits better with compile_str.
>
> However I don't have that much of a strong opinion either, as long as we
> make sure to always pass it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/2765/files#r28447443>.
>